.github/actions/templates/getWorkflowInput/scripts/Get-WorkflowDefaultInput.ps1

```diff
@@ -14,40 +14,115 @@ Get-WorkflowDefaultInput -workflowPath 'path/to/workflow' -verbose
 Retrieve input parameter default values for the 'path/to/workflow' workflow.
 #>
 function Get-WorkflowDefaultInput {
-
+    [CmdletBinding()]
     param (
         [Parameter(Mandatory)]
         [string] $workflowPath
-
     )
 
     begin {
         Write-Debug ('{0} entered' -f $MyInvocation.MyCommand)
+
+        #region Helper Functions
+
+        <#
+        .SYNOPSIS
+        Retrieve indentation of a line.
+
+        .PARAMETER Line
+        Mandatory. The line to analyse for indentation.
+
+        .EXAMPLE
+        $Line = '    Test'
+        Get-LineIndentation -Line $Line
+        4
+
+        Retrieve indentation of a line.
+        #>
+        function Get-LineIndentation {
+            [CmdletBinding()]
+            param (
+                [Parameter()]
+                [string] $Line
+            )
+            begin {}
+            process {
+                $indentation = 0
+                for ($i = 0; $i -lt $Line.Length; $i++) {
+                    $Char = $Line[$i]
+                    switch -regex ($Char) {
+                        '`t' {
+                            $indentation += 2
+                        }
+                        ' ' {
+                            $indentation += 1
+                        }
+                        default {
+                            return $indentation
+                        }
+                    }
+                }
+                return $indentation
+            }
+            end {}
+        }
+
+        <#
+        .SYNOPSIS
+        Retrieve default value for a specified input in a workflow.
+
+        .PARAMETER InputName
+        Mandatory. The name of the input to get the default value for.
+
+        .PARAMETER Content
+        Mandatory. The content of the GitHub workflow file.
+
+        .EXAMPLE
+        $content = Get-Content -Path .\workflow.yml
+        Get-DefaultValue -Text 'removeDeployment' -Content $Content
+
+        Retrieve input default values for the 'removeDeployment' in the workflow.yml file.
+        #>
+        function Get-DefaultValue {
+            [CmdletBinding()]
+            param (
+                [Parameter(Mandatory)]
+                [string] $InputName,
+                [Parameter(Mandatory)]
+                [string[]] $Content
+            )
+            $Content = $Content.Split([Environment]::NewLine)
+            $SectionIndex = ((0..($Content.Count - 1)) | Where-Object { $Content[$_] -match "$InputName" })[0]
+            $SectionIndentation = Get-LineIndentation -Line $Content[$SectionIndex]
+            $LineIndentation = $SectionIndentation
+            for ($i = $SectionIndex + 1; $true; $i++) {
```

Comment:
Something tells me we should not use `$true` here. Unlikely that this loop will never conclude, but I'd much rarther also count until `$Content.Count-1`

Reply:
Fixed